core/controllers/domain_objects_validator.py

```diff
@@ -86,17 +99,50 @@ def validate_change_dict_for_blog_post(change_dict):
             raise Exception(
                 'Invalid tags provided. Tags not in default tags list.')
 
+    return change_dict
+
 
 def validate_collection_change(obj):
     # type: (Dict[String, Any]) -> None
     """Validates collection change.
 
     Args:
         obj: dict. Data that needs to be validated.
+
+    Returns:
+        CollectionChange. Returns CollectionChange object after validation.
     """
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
-    collection_domain.CollectionChange(obj)
+    return collection_domain.CollectionChange(obj)
+
+
+def validate_task_entry_for_improvements(task_entry_dict):
```

Comment:
Same thing: the method name is wrong; the docstring is completely wrong. Please proofread carefully.

Reply:
Updated.